File: analyzer/src/main/java/pl/kwako/metroid_map/analyzer/MapAnalyzer.java
package pl.kwako.metroid_map.analyzer;

import pl.kwako.metroid_map.analyzer.region.RegionSettings;
import pl.kwako.metroid_map.persistence.ObjectRepository;

class MapAnalyzer {

    private final Stage1 stage1;
    private final ObjectRepository objectRepository;

    public MapAnalyzer(Stage1 stage1, ObjectRepository objectRepository) {
        this.stage1 = stage1;
        this.objectRepository = objectRepository;
    }

    public void parseRegionStage1(RegionSettings regionSettings) {
        var stage1Result = stage1.runStage1(regionSettings);
        var filePath = String.format("analyze/%s/stage01_result.json", regionSettings.getImageFileName());
        objectRepository.toFile(stage1Result, filePath);
    }
}

File: analyzer/src/main/java/pl/kwako/metroid_map/analyzer/MapAnalyzer.kt
package pl.kwako.metroid_map.analyzer

import pl.kwako.metroid_map.persistence.ObjectRepository
import java.util.logging.Level
import java.util.logging.Logger

internal class MapAnalyzer(private val stage1: Stage1,
                           private val objectRepository: ObjectRepository) {

    fun analyzeFile(fileName: String) {
        Logger.getGlobal().log(Level.INFO, "START $fileName")
        val regionSettings = objectRepository.fromResource(fileName, RegionSettings::class.java)
        val stage1Result = stage1.runStage1(regionSettings)
        val filePath = String.format("analyze/%s/stage01_result.json", regionSettings.imageFileName)
        objectRepository.toFile(stage1Result, filePath)
        Logger.getGlobal().log(Level.INFO, "END $fileName")
    }
}